Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+
+#include "ObjCXamlControls.h"
+#include "Dwrite_3.h"
+
+using namespace Microsoft::WRL;
+using namespace Platform;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Text;
+using namespace Windows::UI::Xaml::Media;
+
+namespace UIKit {
+namespace Xaml {
+
+DependencyProperty^ TextField::s_textProperty = nullptr;
+DependencyProperty^ TextField::s_secureEntryProperty = nullptr;
+DependencyProperty^ TextField::s_placeholderProperty = nullptr;
+DependencyProperty^ TextField::s_inputScopeProperty = nullptr;
+DependencyProperty^ TextField::s_enabledProperty = nullptr;
+DependencyProperty^ TextField::s_foregroundProperty = nullptr;
+DependencyProperty^ TextField::s_textAlignmentProperty = nullptr;
+DependencyProperty^ TextField::s_textVerticalAlignmentProperty = nullptr;
+DependencyProperty^ TextField::s_borderStyleProperty = nullptr;
+
+unsigned int TextField::c_borderCornerRadius = 8;
+
+Platform::Boolean TextField::s_dependencyPropertiesRegistered = false;
+// TOOD: 
+template <typename T>
+T TextField::FindTemplateChild(FrameworkElement^ source, Platform::String^ name) {
+    T target = nullptr;
+    if (source) {
+        unsigned int count = VisualTreeHelper::GetChildrenCount(source);
+        if (count > 0) {
+            auto templateRoot = dynamic_cast<FrameworkElement^>(VisualTreeHelper::GetChild(source, 0));
+            if (templateRoot) {
+                target = dynamic_cast<T>(templateRoot->FindName(name));
+            }
+        }
+    }
+
+    return target;
+}
+
+TextField::TextField() {
+    _RegisterDependencyProperties();
+
+    InitializeComponent();
+
+    // by default control is enabled
+    Enabled = true;
+
+    // by default control is Non-secure
+    SecureEntry = false;
+
+    // by default no border
+    BorderStyle = TextBorderStyle::BorderStyleNone;
+
+    // by default vertical center aligned
+    VerticalAlignment = Windows::UI::Xaml::VerticalAlignment::Center;
+}
+
+void TextField::_SwitchToSecureMode() {
+    if (_backingControl) {
+        // switch to SecureMode, check if backingControl is a textBox
+        // remove all listeners before we re-assign a password to it
+        auto textBox = dynamic_cast<Controls::TextBox^>(_backingControl);
+        if (textBox) {
+            textBox->TextChanged -= _textChangedHandlerRegistrationToken;
+            textBox->GotFocus -= _gotFocusHandlerRegistrationToken;
+            textBox->LostFocus -= _lostFocusHandlerRegistrationToken;
+            textBox->KeyDown -= _keydownHandlerRegistrationToken;
+        }
+
+        _textChangedHandlerRegistrationToken = {0};
+        _gotFocusHandlerRegistrationToken = {0};
+        _lostFocusHandlerRegistrationToken = {0};
+        _keydownHandlerRegistrationToken = {0};
+        
+        unsigned int index = 0;
+        if (Children->IndexOf(_backingControl, &index)) {
+            Children->RemoveAt(index);
+        }
+
+        _backingControl = nullptr;
+    }
+
```

Comment:
The selected code could easily pulled out into a separate function and reused--all the APIs are on control, or it could be combined with the next function since much of the code is similar.